pkg/mcs/exportspec.go

```diff
@@ -139,13 +164,20 @@ func (es *ExportSpec) IsPrefferredOver(another *ExportSpec) bool {
 // Retruns true when the specifications are compatible, otherwise returns
 // false and the name of the conflicting field.
 // @todo do we want to collect a map of the conflicting Global Properties?
-func (es *ExportSpec) IsCompatibleWith(another *ExportSpec) (bool, string) {
+func (es *ExportSpec) IsCompatibleWith(another *ExportSpec) *CompatibilityError {
 	if es.Service.Type != another.Service.Type {
-		return false, "type"
+		return &CompatibilityError{another.ClusterID, TypeField}
 	} else if es.Service.SessionAffinity != another.Service.SessionAffinity {
-		return false, "affinity"
+		return &CompatibilityError{another.ClusterID, AffinityField}
 	} else if !reflect.DeepEqual(es.Service.SessionAffinityConfig, another.Service.SessionAffinityConfig) {
-		return false, "affinityConfig"
+		return &CompatibilityError{another.ClusterID, AffinityConfigField}
+	}
+
+	if es.Name != another.Name {
```

Comment:
Pull these up as the first checks, they are more coarse grained (i.e., not the same export so no need to compare field by field)